ethcore/wasm/src/runtime.rs

```diff
@@ -102,6 +103,13 @@ pub struct RuntimeContext {
 	pub value: U256,
 }
 
+macro_rules! charge {
```

Comment:
Why not a simple function?

Reply:
because of double borrowing of self

Reply:
ugh, ok

Reply:
seems like a `fn charge<F>(&mut self, f: F) where F: FnOnce(&Schedule) -> u64` would work in all cases here